functions.go

```diff
@@ -108,3 +158,79 @@ func (s *FunctionsServiceOp) DeleteNamespace(ctx context.Context, namespace stri
 	}
 	return resp, nil
 }
+
+// ListTriggers gets a list of triggers
+func (s *FunctionsServiceOp) ListTriggers(ctx context.Context, namespace string) ([]FunctionsTrigger, *Response, error) {
+	path := fmt.Sprintf(functionsTriggerBasePath, namespace)
+	req, err := s.client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	root := new(triggersRoot)
+	resp, err := s.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+	return root.Triggers, resp, nil
+}
+
+// GetTrigger gets a single trigger
+func (s *FunctionsServiceOp) GetTrigger(ctx context.Context, namespace string, trigger string) (*FunctionsTrigger, *Response, error) {
+	path := fmt.Sprintf(functionsTriggerBasePath+"/%s", namespace, trigger)
+	req, err := s.client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	root := new(triggerRoot)
+	resp, err := s.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+	return root.Trigger, resp, nil
+}
+
+// CreateTrigger creates a trigger
+func (s *FunctionsServiceOp) CreateTrigger(ctx context.Context, namespace string, opts *FunctionsTriggerCreateRequest) (*FunctionsTrigger, *Response, error) {
+	path := fmt.Sprintf(functionsTriggerBasePath, namespace)
```

Comment:
should probably validate the `CreateRequest` here to ensure appropriate fields are present for the provided `type`? Eg. `type: scheduled` should have the `schedule_details` field populated.

Reply:
I have not seen this pattern in godo, the validation will be done by the consumer, or the API itself.

Reply:
Like @ddebarros mentioned, we generally let the API itself do the validation. It allows us to be a bit more flexible. E.g. a default change or making something non-required can be done at the API layer without requiring clients to update.